src/components/TransactionMenu/index.tsx

```diff
@@ -50,29 +88,81 @@ export default function TransactionMenu({
 
   // Menu Item for a Transaction
   function TransactionMenuItem(transaction: Transaction) {
-    let color: string;
-
-    switch (transaction.status) {
-      case 'confirming':
-        color = 'orange';
-        break;
-      case 'success':
-        color = 'green';
+    let color: string = 'orange';
+
+    if (transaction.confirmations >= 12) {
+      color = 'green';
     }
 
     return (
-      <MenuItem onClick={handleClose}>
+      <MenuItem>
         <Chip
           avatar={
             <small style={{ marginRight: '10em', color: color }}>
-              {transaction.status}({transaction.confirmations})
+              ({transaction.confirmations} confirmations)
             </small>
           }
           label={shortenWalletAddress(transaction.hash)}
         />
       </MenuItem>
     );
   }
+
+  // Modal content
+  const modalChildren = (
+    <div>
+      <h3>Transactions</h3>
+      <TableContainer component={Paper}>
+        <Table aria-label="simple table">
+          <TableHead>
+            <TableRow>
+              <TableCell align="right">#</TableCell>
+              <TableCell align="left">From -&gt; To</TableCell>
+              <TableCell align="left">Confirmations</TableCell>
+              <TableCell align="left">Amount</TableCell>
+              <TableCell align="center">Remove</TableCell>
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {net.transactions.map((t) => (
+              <TableRow key={t.hash}>
+                <TableCell component="th" scope="row" align="left">
+                  <Link href={`https://ropsten.etherscan.io/tx/${t.hash}`}>
+                    {t.hash}
+                  </Link>
+                </TableCell>
+                <TableCell align="left">
+                  <Link
+                    href={`https://ropsten.etherscan.io/address/${t.sender}`}
+                  >
+                    <small>{shortenWalletAddress(t.sender)} </small>
+                  </Link>
+                  -&gt;
+                  <small> {shortenWalletAddress(t.receiver)}</small>
+                </TableCell>
+                <TableCell align="left">
+                  {t.confirmations >= 12 ? (
+                    <span style={{ color: 'green' }}>{t.confirmations}</span>
+                  ) : (
+                    <span style={{ color: 'red' }}>{t.confirmations}</span>
+                  )}
+                </TableCell>
+                <TableCell align="left">
+                  <small>{t.amount}</small>
+                </TableCell>
+                <TableCell align="center">
+                  <button onClick={handleDelete} data-hash={t.hash}>
+                    Delete
```

Comment:
whats the purpose of this delete button?

Reply:
It removes a transaction from the array of all submitted transactions, it is not required just added it as an ux feature.

Reply:
hm i'm not sure we should have any clear/delete functionality at all - don't think it necessarily provides user value, and could also be confusing in terms of what it means after refresh/coming back to app.

Reply:
Makes sense. I removed the methods